modules/apps/asset/asset-publisher-web/src/main/resources/META-INF/resources/view_asset_entries_abstract.jsp

```diff
@@ -266,45 +266,41 @@ for (AssetEntry assetEntry : assetEntryResult.getAssetEntries()) {
 						>
 
 							<%
-							String id = assetEntry.getEntryId() + StringUtil.randomId();
 							String label = LanguageUtil.format(request, "print-x", HtmlUtil.escape(title));
+
+							String printPageURL = PortletURLBuilder.createRenderURL(
+								renderResponse
+							).setMVCPath(
+								"/view_content.jsp"
+							).setParameter(
+								"assetEntryId", assetEntry.getEntryId()
+							).setParameter(
+								"languageId", LanguageUtil.getLanguageId(request)
+							).setParameter(
+								"type", assetRendererFactory.getType()
+							).setParameter(
+								"viewMode", Constants.PRINT
+							).setWindowState(
+								LiferayWindowState.POP_UP
+							).buildString();
 							%>
 
 							<clay:button
+								additionalProps='<%=
+									HashMapBuilder.<String, Object>put(
+										"printPageURL", printPageURL
+									).build()
+								%>'
 								aria-label="<%= label %>"
 								borderless="<%= true %>"
 								displayType="secondary"
 								icon="print"
-								onClick='<%= "javascript:" + liferayPortletResponse.getNamespace() + "printPage_" + id + "();" %>'
+								id='<%= liferayPortletResponse.getNamespace() + "printPageButton" %>'
+								propsTransformer="js/printPageButtonPropsTransformer"
```

Comment:
I believe a propsTransformer should _return_ some new props via object right? So instead of having to use an event listener, I think you can  return `{onClick: //...` in that file and that gets set as the prop of the component.


It has been a little while since I have dug into this, but I don't think you need to be manually setting event listeners